### PR TITLE
feat/wf-service-proj-scope

### PR DESCRIPTION
**Pull request by chesterkmr**
### Description
- Added ESLint plugin which lints repositories without projectIds
- Updated all code dependant on repositories (Controllers/Services/Middlewares)

### Related issues
 * Provide a link to each related issue.

### Breaking changes
 * Describe the breaking changes that this pull request introduces.

### How these changes were tested
 * Describe the tests that you ran to verify your changes, including devices, operating systems, browsers and versions.

### Examples and references
 * Links, screenshots, and other resources related to this change.

### Checklist
- [] I have read the [contribution guidelines](CONTRIBUTING.md) of this project
- [] I have read the [style guidelines](STYLE_GUIDE.md) of this project
- [] I have performed a self-review of my own code
- [] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [] My changes generate no new warnings and errors
- [] New and existing tests pass locally with my changes

### CODE REVIEW

**Review by Omri-Levy — CHANGES_REQUESTED** [review]
`eslint-plugin-ballerine` should be a package and not in `workflows-service`.

File: services/workflows-service/.eslintrc.cjs [hunk 1]
```diff
@@ -1,4 +1,5 @@
 const config = require('../../packages/config/eslintrc.base.cjs');
+const checkRepositoryArgsPlugin = require('./plugins/check-repository-args');
```
Comment (Omri-Levy) [inline services/workflows-service/.eslintrc.cjs]: See if we can improve the plugin's name, the plugin is related to `projectId` and multi-tenancy.

File: services/workflows-service/package.json [hunk 2]
```diff
@@ -120,6 +120,7 @@
     "eslint": "^8.35.0",
     "eslint-config-prettier": "^8.7.0",
     "eslint-import-resolver-typescript": "^3.5.3",
+    "eslint-plugin-ballerine": "link:plugins/check-repository-args",
```
Comment (Omri-Levy) [inline services/workflows-service/package.json]: Is `link` intentional here?

File: services/workflows-service/plugins/check-repository-args/check-repository-args.plugin.js [hunk 3]
```diff
@@ -0,0 +1,6 @@
+// eslint-disable-next-line
```
Comment (Omri-Levy) [inline services/workflows-service/plugins/check-repository-args/check-repository-args.plugin.js]: Change file to `.cjs`

File: services/workflows-service/plugins/check-repository-args/check-repository-args.rule.js
Comment (Omri-Levy) [inline services/workflows-service/plugins/check-repository-args/check-repository-args.rule.js:1]: Change file to `.cjs`.

File: services/workflows-service/plugins/check-repository-args/check-repository-args.rule.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+    return {
+      MethodDefinition: function (node) {
+        if (!isRepository || node.key.name === 'constructor') return;
+        const isUnscoped = node.key.name.toLowerCase().includes('unscoped');
+
+        if (isUnscoped) return;
+
+        const isProjectIdsIncluded =
+          node.value.params.length &&
```
Comment (Omri-Levy) [inline services/workflows-service/plugins/check-repository-args/check-repository-args.rule.js]: Can be simplified, length check is needed for `every` and not `some`.

File: services/workflows-service/src/auth/jwt/jwt.strategy.ts [hunk 5]
```diff
@@ -18,7 +18,7 @@ export class JwtStrategy extends PassportStrategy(Strategy) implements IAuthStra
 
   async validate(payload: UserInfo): Promise<UserInfo> {
     const { id } = payload;
-    const user = await this.userService.getByIdUnscoped(id);
+    const user = await this.userService.getByIdUnscoped(id, {}, null);
```
Comment (Blokh) [inline services/workflows-service/src/auth/jwt/jwt.strategy.ts]: this is unscoped.
why did it change?

File: services/workflows-service/src/auth/session-serializer.ts [hunk 6]
```diff
@@ -28,15 +28,19 @@ export class SessionSerializer extends PassportSerializer {
...
   ) {
     try {
-      const userResult = await this.userService.getByIdUnscoped(user.user!.id!, {
-        select: {
-          id: true,
-          email: true,
-          firstName: true,
-          lastName: true,
-          userToProjects: { select: { projectId: true } },
+      const userResult = await this.userService.getByIdUnscoped(
```
Comment (Blokh) [inline services/workflows-service/src/auth/session-serializer.ts]: this is unscoped.
why did it change?

File: services/workflows-service/src/business/business.controller.external.ts [hunk 7]
```diff
@@ -83,7 +92,7 @@ export class BusinessControllerExternal {
   @UseCustomerAuthGuard()
   async getById(@common.Param() params: BusinessWhereUniqueInput): Promise<BusinessModel | null> {
     try {
-      const business = await this.service.getById(params.id);
+      const business = await this.service.getById(params.id, {});
```
Comment (Blokh) [inline services/workflows-service/src/business/business.controller.external.ts]: why is this unscoped?
it should be scoped.

File: services/workflows-service/plugins/check-repository-args/check-repository-args.rule.js (hunk 4 above)
Comment (chesterkmr) [reply services/workflows-service/plugins/check-repository-args/check-repository-args.rule.js]: Good point,thanks

**Pull request merged**